server/middleware/workstack.js

```diff
@@ -66,14 +66,26 @@ function workstackApiResponseMiddleware(req, res) {
     res.json(res.locals.workstack);
 }
 
+const updateCaseDataMoveTeamRequest = async (req, res, [endpoint, body, headers]) => {
+    const logger = getLogger(req.requestId);
+    try {
+        await workflowService.put(endpoint, body, headers);
```

Comment:
This seems to be a bit of a waste of an `async`/`await` -- this is essentially blocking anyway? (I'm never sure how async/await works).

Reply:
I think it's because of the way we handle middleware. This is currently used in a rather funky way but could be used elsewhere, where we need to wait for it to finish before progressing further middleware.

Reply:
We need this to handle the promise rejection on failure. I'll put this down to a limitation with our implementation. I do agree though, we should be handling this on a higher level.